x-pack/plugins/maps/common/descriptor_types/map_descriptor.ts

```diff
@@ -63,8 +63,13 @@ export type TooltipState = {
 
 export type DrawState = {
   actionId: string;
-  drawType: DRAW_TYPE;
+  drawShape?: DRAW_SHAPE;
   filterLabel?: string; // point radius filter alias
   geometryLabel?: string;
   relation?: ES_SPATIAL_RELATIONS;
 };
+
+export type EditState = {
+  layerId?: string;
```

Comment:
Thanks for doing all the work of making layerId required. Now you just need to change `layerId?: string;` to `layerId: string;`